src/error.rs

```diff
@@ -38,10 +38,19 @@ pub enum Error {
 impl fmt::Display for Error {
     fn fmt(&self, fmt: &mut fmt::Formatter<'_>) -> fmt::Result {
         match self {
-            Error::RpcFailure(RpcStatus { status, details }) => match details {
-                Some(details) => write!(fmt, "RpcFailure: {} {}", status, details),
-                None => write!(fmt, "RpcFailure: {}", status),
-            },
+            Error::RpcFailure(s) => {
+                if s.error_details().is_empty() && s.error_message().is_empty() {
+                    write!(fmt, "RpcFailure: {}", s.error_code())
+                } else {
+                    write!(
+                        fmt,
+                        "RpcFailure: [{}] binary: {} {}",
+                        s.error_code(),
+                        !s.error_details().is_empty(),
```

Comment:
that `binary: true/false` part is quite obscure to people reading the error message

Reply:
Any suggestions?

Reply:
I think we should simply ignore it (keep the old `Display` output). Saying that `.error_details()` exists mean nothing in the log since you can't retrieve the details again, while printing the entire `.error_details()` into the log is too noisy.